core/new-gui/src/app/workspace/component/result-panel/result-panel.component.ts

```diff
@@ -35,7 +39,11 @@ export class ResultPanelComponent {
 
   private static readonly PRETTY_JSON_TEXT_LIMIT: number = 50000;
   private static readonly TABLE_COLUMN_TEXT_LIMIT: number = 1000;
-
+  public result :ReadonlyArray<ResultObject> | undefined;
+  public resultMap: Map<string, Result> = new Map<string, Result>();
```

Comment:
If I understand correctly, `resultMap` holds all the execution result of all sinks. where `result` hold the result of the currently selected sink. 
move `resultMap` to `ExecuteWorkflowService` and keep `result` in this component. The reason is we try to keep the global states that might be shared in the service, and keep local states that is only used in the component itself in the component. A component only hold the states needed to display.

In this case, I think `resultMap` can be viewed as a global state. and `result` is the result of the currently selected operator. Whenever operator selection is changed, update the local `result` variable by fetching it from service.